### ISSUE TITLE
False positive finding for E0605: Invalid format for __all__

### ISSUE THREAD

**Issue by kasium** [PyCQA/pylint#4711]
### Steps to reproduce
Given a file `__init__.py`:
```python
from foo import bar
__all__ = list(globals().keys())
```

### Current behavior
Result of `pylint __init__.py`:
```
configuration:1:0: E0605: Invalid format for __all__, must be tuple or list (invalid-all-format)
```

### Expected behavior
No finding. The attribute is a list at runtime but I guess the check doesn't see this

### pylint --version output
Result of `pylint --version` output:
```
pylint 2.9.3
astroid 2.6.2
Python 3.7.1 (default, Oct 22 2018, 13:16:18) [GCC]
```

**Comment by kasium** [comment]
So I did some troubleshooting. The type of the list at pylint time is `<Instance of builtins.list at 0x140612117943968>`

So, the check fails here: https://github.com/PyCQA/pylint/blob/741276e90a3698c8bf5ac621669453b67b7c2c7c/pylint/checkers/variables.py#L2013

**Comment by Pierre-Sassoulas** [comment]
If I understand what you're saying the fix could be : `if not isinstance(assigned, (astroid.Tuple, astroid.List, list)): ` ? Could you try to open a merge request and add a functional test in `tests/functional/` to get the authorship of your fix ? :)

**Comment by kasium** [comment]
I guess that could be the fix 😄 . Let me give it a try

**Comment by kasium** [comment]
@Pierre-Sassoulas bad new, still not fixed.

I'm not sure, but it seems, that these "_valid_x.py" files are not running in the tests at all. Can you please check.

Also , the module name of the finding is false:
```
************* Module builtins
configuration:1:0: E0605: Invalid format for __all__, must be tuple or list (invalid-all-format)
```
The module `builtins` and filename `configuration` make no sense. I guess that this code 
https://github.com/PyCQA/pylint/blob/741276e90a3698c8bf5ac621669453b67b7c2c7c/pylint/checkers/variables.py#L2014
should have `node=node`.

**Comment by Pierre-Sassoulas** [comment]
Thank you for finding that bug with the bad node being given. The valid_x file are all being executed but the example given in this issue is not tested in it. It's still giving:
```
************* Module invalid_all_format_valid_5
tests/functional/i/invalid/invalid_all_format_valid_5.py:1:0: E0605: Invalid format for __all__, must be tuple or list (invalid-all-format)
```
See https://github.com/PyCQA/pylint/pull/4953

**Comment by kasium** [comment]
Thanks a lot for the quick help and the PR. It would be great to see this in a 2.10.x release 😄

**Comment by kasium** [comment]
So, the actual type is an astroid instance object which represents `builtins.list`. Not sure how to check for that.
One option would be to use `assigned.pytype() == 'builtins.list'`. What do you think?

**Comment by Pierre-Sassoulas** [comment]
You're right, I also handled tuple the same way.

**Comment by kasium** [comment]
Great 😄
